src/MapClient.ts

```diff
@@ -75,12 +77,12 @@ export class InnerMapClient<T extends any> implements ObjectClient {
     return Object.keys(this.store);
   }
 
-  get(key: string): T {
-    return this.store[key] as T;
+  get<K extends keyof T>(key: K): T {
```

Comment:
what is the point of adding these generics? is it possible for a key to not be a string?

Reply:
Oh! It's to prevent someone from adding the _wrong_ string:

```tsx
interface Bike {
  wheels: number,
  name: string
}

const map = room.map<Bike>("bike")

map.get("wheels") // OK
map.get("door").  // TypeError: 'door' does not exist on interface 'Bike'
```

The phrase `K extends keyof T` means the union of the keys of an object. These two are similar:
```ts
type K extends keyof Bike == "wheels" | "name"
```

But since this `K` type we've made is used to index `T`, we also get types of the values as well! :D

```tsx
map.get("wheels") // type is number  
map.get("name")   // type is string
```

Reply:
Oh that seems really nice for interfaces! Is there an easy way to also specify a map with arbitrary string keys?

Reply:
Oh would you just use `<any>`?

Reply:
So you could use `<any>`, but a more powerful version is this:

```js
interface Car {
  [key: string]: string
}
```